internal/worker/worker.go

```diff
@@ -50,8 +52,14 @@ func (w *worker) LoadKmod(ctx context.Context, cfg *kmmv1beta1.ModuleConfig) err
 		}
 	}
 
-	// TODO copy firmware
-	// TODO handle ModulesLoadingOrder
+	// prepare firmware
+	if cfg.Modprobe.FirmwarePath != "" {
+		baseImageDir := pr.fsDir + "/" + cfg.Modprobe.FirmwarePath
+		w.logger.Info("preparing fimrware for loading", "image directory", baseImageDir, "host mount directory", firmwareMountPath)
+		if err = cp.Copy(baseImageDir, firmwareMountPath); err != nil {
```

Comment:
What this function does in case `baseImageDir` exists is not very clear:
```
	// Merge preserves or overwrites existing files under the dir (default behavior).
```
https://pkg.go.dev/github.com/otiai10/copy#DirExistsAction
When does it replace and when does it overwrite?
Also, WDYT about using the `OnError` option to log copy errors that may occur?

Reply:
You probably mean in case there are already directories/files exist at destination, right? I have looked at the copy package code and run a few tests, and i think we are ok: if directory already exists, then new files will be added,  files with the same name will be overwritten, files present only at dest will be kept as they are. 
Also, added onError function